eth/catalyst/simulated_beacon.go

```diff
@@ -163,10 +176,14 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal) error {
 	if payload.Number%devEpochLength == 0 {
 		finalizedHash = payload.BlockHash
 	} else {
-		finalizedHash = c.eth.BlockChain().GetBlockByNumber((payload.Number - 1) / devEpochLength * devEpochLength).Hash()
+		if block := c.eth.BlockChain().GetBlockByNumber((payload.Number - 1) / devEpochLength * devEpochLength); block != nil {
+			finalizedHash = block.Hash()
+		} else {
+			finalizedHash = payload.BlockHash
```

Comment:
Please add a comment about why this is correct / what leads to this happening. If one just eyes through the code, it looks pretty strange to set `finalizedHash` here like this.

Reply:
I think the only reason that `GetBlockByNumber` could return `nil` here is if a `setHead` were to occur between line 149 and here (and set the head > 32 blocks behind).  In this case, I think we want to return early from this function instead of setting `finalizedHash = payload.BlockHash`.

Reply:
@jwasinger Sorry, I didn't understand correctly before, please help review again